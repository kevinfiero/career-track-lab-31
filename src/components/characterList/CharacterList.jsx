import React from 'react';
import PropTypes from 'prop-types';
import CharacterListItem from './CharacterListItem';

const CharacterList = ({ characters }) => {
  const characterElements = characters.map(character => (
    <li key={character.id}>
      <CharacterListItem image={character.image} name={character.name} id={character.id} />
    </li>
  ));

  return (
    <ul data-testid="characters">
      {characterElements}
    </ul>
  );
};

CharacterList.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  })).isRequired
};

export default CharacterList;
