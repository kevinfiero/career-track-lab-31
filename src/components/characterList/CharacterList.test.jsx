import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import CharacterList from './CharacterList';

describe('CharacterList component', () => {
  afterEach(() => cleanup());
  it('renders CharacterList', () => {
    const { asFragment } = render(
      <Router>
        <CharacterList 
          characters = 
            {[
              {
                image:'https://vignette.wikia.nocookie.net/heyarnold/images/f/f5/Arnold.png/revision/latest?cb=20181027162333',
                name:'Arnold Shortman',
                id:'5da237699734fdcb7bef8f51'
              },
              {
                image:'https://vignette.wikia.nocookie.net/heyarnold/images/7/79/GrandpaPhil.png/revision/latest/scale-to-width-down/300?cb=20160924030436',
                name:'Grandpa Phil',
                id:'5da237699734fdcb7bef8f52'
              }
            ]}
        />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
