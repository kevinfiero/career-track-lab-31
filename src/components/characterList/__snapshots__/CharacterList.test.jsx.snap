// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component renders CharacterList 1`] = `
<DocumentFragment>
  <ul
    data-testid="characters"
  >
    <li>
      <a
        href="/5da237699734fdcb7bef8f51"
      >
        <figure>
          <img
            src="https://vignette.wikia.nocookie.net/heyarnold/images/f/f5/Arnold.png/revision/latest?cb=20181027162333"
          />
          <figcaption>
            <h1>
              Arnold Shortman
            </h1>
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5da237699734fdcb7bef8f52"
      >
        <figure>
          <img
            src="https://vignette.wikia.nocookie.net/heyarnold/images/7/79/GrandpaPhil.png/revision/latest/scale-to-width-down/300?cb=20160924030436"
          />
          <figcaption>
            <h1>
              Grandpa Phil
            </h1>
          </figcaption>
        </figure>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
